src/libraries/System.Data.Common/src/System/Xml/XmlDataDocument.cs

```diff
@@ -2967,14 +2964,14 @@ private void AssertNonLiveRows(XmlNode node)
             IsFoliationEnabled = false;
             try
             {
-                XmlBoundElement rowElement = node as XmlBoundElement;
+                XmlBoundElement? rowElement = node as XmlBoundElement;
                 if (rowElement != null && rowElement.Row != null)
                     Debug.Assert(!IsRowLive(rowElement.Row));
                 TreeIterator iter = new TreeIterator(node);
                 for (bool fMore = iter.NextRowElement(); fMore; fMore = iter.NextRowElement())
                 {
                     rowElement = iter.CurrentNode as XmlBoundElement;
-                    Debug.Assert(rowElement.Row != null);
+                    Debug.Assert(rowElement!.Row != null);
                     Debug.Assert(!IsRowLive(rowElement.Row));
                 }
             }
```

Comment:
On line 2984: base method returns `XmlElement?`, I think we should honor nullability regardless of throwing.

Reply:
Fixed